src/main/java/org/sakaiproject/elfinder/controller/executors/SakaiRenameCommandExecutor.java

```diff
@@ -0,0 +1,30 @@
+package org.sakaiproject.elfinder.controller.executors;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+
+import org.json.JSONObject;
+
+import cn.bluejoe.elfinder.controller.executor.AbstractJsonCommandExecutor;
+import cn.bluejoe.elfinder.controller.executor.CommandExecutor;
+import cn.bluejoe.elfinder.controller.executor.FsItemEx;
+import cn.bluejoe.elfinder.service.FsService;
+
+public class SakaiRenameCommandExecutor extends AbstractJsonCommandExecutor implements CommandExecutor
+{
+	@Override
+	public void execute(FsService fsService, HttpServletRequest request, ServletContext servletContext, JSONObject json)
+			throws Exception
+	{
+		String target = request.getParameter("target");
+		String current = request.getParameter("current");
+		String name = request.getParameter("name");
+
+		FsItemEx fsi = super.findItem(fsService, target);
+		FsItemEx dst = new FsItemEx(fsi.getParent(), name);
+		fsi.renameTo(dst);
+
+		json.put("added", new Object[] { getFsItemInfo(request, fsi) });
```

Comment:
So you're rename isn't doing a rename, it's changing the display name of the file.